app/gimletlet/app-mgmt.toml

```diff
@@ -85,7 +85,7 @@ path = "../../drv/user-leds"
 name = "drv-user-leds"
 features = ["stm32h7"]
 priority = 2
-requires = {flash = 2048, ram = 1024}
```

Comment:
This seems very suspicious – the `user-leds` task should be _tiny_, so this would be a good starting place to investigate the size changes.

Reply:
99% of the changes in size are on the order of 100-200 bytes; only one of them was around 1000 bytes.

The issue is that these must be powers of two, and so if you need 2049 bytes of flash, you end up needing to put in 4096, even if almost all of it is wasted space.

Reply:
I did some detective work (`arm-none-eabi-objdump -d --demangle -t user_leds | grep " F .text" | sort -k 6 | cut -w -f5-`) and found that the main change is in `core::fmt::Formatter::pad`, which went from 690 bytes to 1970 bytes!

Here's the full set of functions that changed size
```diff
4c4
< 000002b2      core::fmt::Formatter::pad
---
> 000007b2      core::fmt::Formatter::pad
13d12
< 00000014      drv_user_leds::idl::InOrderUserLedsImpl::closed_recv_fail
15c14
< 00000208      main
---
> 0000020c      main
```

I found [this Rust commit](https://github.com/rust-lang/rust/commit/6c9e708f4be04a5da9267171602be135789b12b8) which touched the function recently, but dunno if it's worth digging further...